gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -355,6 +361,11 @@ public GoogleCloudStorageImpl(
 
     this.storageRequestAuthorizer = initializeStorageRequestAuthorizer(storageOptions);
     this.downscopedAccessTokenFn = downscopedAccessTokenFn;
+
+    // Set GSC connector logging-level based on the passed debug logging option.
+    if (googleLogger.getLevel() == null) {
```

Comment:
What if the level is non-`null`? I thought the property should override it

Reply:
The intention is to only configure this value on initialization, If the value is not null then it's likely the user is manually overriding the level which it's better to not force it back.

This is also a check to avoid repeatedly calling a synchronized `setLevel` method.

Reply:
Edit: I re-wrote this to only override when the level value doen't match with the config value